### ISSUE TITLE
Tree scan corner case.

### ISSUE THREAD

**Issue by rogerioacp** [rogerioacp/SOE#4]
In both equality and range queries, a few search keys are not find the correct value. The problem appears because we have simplified the tree search for the PRF integration and removed the internals state of the tree scan. In particular, when a parent tree node A points to a block B in a lower level  but the element being searched is in the block B+1 then the search fail. This is easily solved by restoring the scan iteration on the server side and move the search to the right leaf.
This is already done in older versions without the PRF. 

Basically the problem appears because the leafs have a right link that sometimes have to be searched to find an element,  and this is address with the _bt_next function in the btree search. However, the current version of tree search is only using the _bt_first function to find the elements to quickly integrate the PRF version. As this only has a small impact on the overall query latency, than it is not supported on the current tests.
